frontend/src/components/Permissions.tsx

```diff
@@ -1,50 +1,119 @@
-import { PlusCircle } from "lucide-react";
-import React, { useEffect, useState } from "react";
+import { format } from "date-fns";
+import { CalendarIcon, PlusCircle, XIcon } from "lucide-react";
+import React, { useState } from "react";
+import Scopes from "src/components/Scopes";
 import { Button } from "src/components/ui/button";
-import { Checkbox } from "src/components/ui/checkbox";
+import { Calendar } from "src/components/ui/calendar";
+import { Input } from "src/components/ui/input";
 import { Label } from "src/components/ui/label";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "src/components/ui/popover";
 import {
   Select,
   SelectContent,
   SelectItem,
   SelectTrigger,
   SelectValue,
 } from "src/components/ui/select";
-import { useCapabilities } from "src/hooks/useCapabilities";
 import { cn } from "src/lib/utils";
 import {
   AppPermissions,
   BudgetRenewalType,
+  NIP_47_PAY_INVOICE_METHOD,
   Scope,
+  WalletCapabilities,
   budgetOptions,
   expiryOptions,
   iconMap,
   scopeDescriptions,
   validBudgetRenewals,
 } from "src/types";
 
+const getTimeZoneDirection = () => {
+  const offset = new Date().getTimezoneOffset();
+
+  return offset <= 0 ? +1 : -1;
+};
+
+const daysFromNow = (date?: Date) => {
+  if (!date) {
+    return 0;
+  }
+  const utcDate = new Date(
+    Date.UTC(
+      date.getUTCFullYear(),
+      date.getUTCMonth(),
+      date.getUTCDate(),
+      23,
+      59,
+      59,
+      0
+    )
+  );
+  return Math.ceil(
+    (new Date(utcDate).getTime() - Date.now()) / (1000 * 60 * 60 * 24)
+  );
+};
+
 interface PermissionsProps {
+  capabilities: WalletCapabilities;
   initialPermissions: AppPermissions;
   onPermissionsChange: (permissions: AppPermissions) => void;
-  budgetUsage?: number;
   canEditPermissions: boolean;
+  budgetUsage?: number;
   isNewConnection?: boolean;
 }
 
 const Permissions: React.FC<PermissionsProps> = ({
+  capabilities,
   initialPermissions,
   onPermissionsChange,
   canEditPermissions,
   isNewConnection,
   budgetUsage,
 }) => {
   const [permissions, setPermissions] = React.useState(initialPermissions);
-  const [days, setDays] = useState(isNewConnection ? 0 : -1);
-  const [expireOptions, setExpireOptions] = useState(!isNewConnection);
-  const { data: capabilities } = useCapabilities();
+  const [expiryDays, setExpiryDays] = useState(
+    daysFromNow(permissions.expiresAt)
+  );
+  const [budgetOption, setBudgetOption] = useState(
```

Comment:
can you please do better with naming and review all the variables in this PR?

Think what this variable does and what it should be called so I understand when I read it